core/controllers/async_download.py

```diff
@@ -0,0 +1,152 @@
+import io
+from datetime import datetime
+from typing import IO
+
+from flask import current_app
+from notifications_python_client.notifications import NotificationsAPIClient
+from werkzeug.datastructures import FileStorage
+
+from config import Config
+from core.aws import _S3_CLIENT
+from core.controllers.download import download
+
+
+def async_download(
+    email_address: str,
+    file_format: str,
+    funds: list[str] | None = None,
+    organisations: list[str] | None = None,
+    regions: list[str] | None = None,
+    rp_start: str | None = None,
+    rp_end: str | None = None,
+    outcome_categories: list[str] | None = None,
+):
+    """Download data, store file in S3 and send an email to the user with the download link."""
+
+    response = download(
+        file_format=file_format,
+        funds=funds,
+        organisations=organisations,
+        regions=regions,
+        rp_start=rp_start,
+        rp_end=rp_end,
+        outcome_categories=outcome_categories,
+    )
+
+    if response.status_code != 200:
+        current_app.logger.error(
+            "Error downloading data from data-store/download endpoint. Status code: {response_status}",
+            extra=dict(response_status=response.status_code),
+        )
+        return
+
+    file_format_str: str = get_file_format_from_extension(file_format)
+    file_size_str: str = get_human_readable_file_size(len(response.data))
+
+    # Upload the file to S3 and get presigned URL
+    file_obj = io.BytesIO(response.data)
+    bucket = Config.AWS_S3_BUCKET_FIND_DATA_FILES
+    find_service_base_url = Config.FIND_SERVICE_BASE_URL
+
+    if not bucket:
+        current_app.logger.error("AWS_S3_BUCKET_FIND_DATA_FILES is not set.")
+    elif not find_service_base_url:
+        current_app.logger.error("FIND_SERVICE_BASE_URL is not set.")
+    else:
```

Comment:
I'd rather the app refused to start up if these vars are missing. Better for the app to not start than for it to start running and be misconfigured (unable to serve this endpoint), I think.

But this might be one to address separately as at the moment we're very forgiving on all of our env vars/config 🤔  I will make a ticket.

Reply:
Removed these logging lines.